File: StoreApplication/python/eth.py
import requests
import time
import datetime

def eth(tsyms):
   qs = { 'fsym': 'ETH', 'tsyms': tsyms.replace(' ', '').upper(), 'sign': True }
   r = requests.get('https://min-api.cryptocompare.com/data/price', params = qs)

   result = { 'data': r.json() }

   if 'Authorization' not in r.headers:
      return None

   sig = r.headers['Authorization'][len('Signature '):]

   for t in sig.split(','):
      val = t.split('"')
      result[val[0][:-1]] = val[1]

   result['timestamp'] = int(time.time())

   return result

if __name__ == '__main__':
   result = eth('sgd ,thb   ,myr')
   for i in result:
      print('%s = %s' %(i, str(result[i])))

File: StoreApplication/javascript/eth.js
const request = require('request');

const URL = 'https://min-api.cryptocompare.com/data/price';

module.exports = function(tsyms = 'SGD') {
	return (
		new Promise((resolve, reject) => {
			request({
				method: 'GET',
				url: URL,
				qs: {
					fsym: 'ETH',
					tsyms: tsyms.toUpperCase().replace(/ /g, ''),
					sign: true
				}
			}, (err, resp, body) => {

				if (err || (!'authorization' in resp.headers)) {
					reject(err || { error: 'Mission Authorization header'});
					return;
				}

				let result = { data: JSON.parse(body) };

   			let sig = resp.headers['authorization'].substring('Signature '.length)
				let terms = sig.split(',')
				for (let t of terms) {
					let val = t.split('"');
					result[val[0].substring(0, val[0].length - 1)] = val[1]
				}

				result['timestamp'] = Math.round((new Date()).getTime() / 1000);

				resolve(result);
			})
		})
	);
}
